### PR TITLE
refactor: Rename key,id,dockey to docID terminology

### LINKED ISSUE #1272: Make `dockey` and `id` request arguments more consistent
Sometimes we use `dockey` and sometimes we use `id` as arguments to request queries/mutations.

See also https://github.com/sourcenetwork/defradb/issues/1467 - it might be worth doing these together.

### LINKED ISSUE #1752: Rename instances of `_key` to `_docID`
Part of https://github.com/sourcenetwork/defradb/issues/1750

Rename instances of `_key` to `_docID`

### PR DESCRIPTION
**Pull request by shahzadlone**
## Relevant issue(s)
- Subtask of #1750 [EPIC]
- [x] Resolves #1752
- [x] Resolves #1272

## BREAKING CHANGE:
- Use of `_key` to access a document's unique id is now deprecated, instead use `_docID`.
- Use of `dockey`/`id` is now deprecated, instead use `docID`.
- Use of `dockeys`/`ids` is now deprecated, instead use `docIDs`.

## Description
- [x] Rename `_key` to `_docID` everywhere.
- [x] Rename `_keys` to `docIDs` in explain the response.
- [x] Rename `_newKey` to `_docIDNew` in backup/recover functionality.
- [x] Fix `_docID` tests.
- [x] Fix explain and backup/recover functionality tests.
- [x] Fix the collectionID order for a P2P test (leaving a note as order was reverted).
- [x] Update all cids.
- [x] Rename all files with `key(s)|dockey(s)` in the name to `doc_id(s)`.
- [x] Document breaking change to pass change detector.

## For Reviewers:
- Main commits to review are the `PR(MAIN)` commits. 
- If you have more time `PR(MINOR) and PR(*TEST)` commits are good to go over

## Disclaimer / Discussion:
I do not like these non-underscored `docID/docIDs`, would be in favor of underscoring these :

```
query {
    User(docIDs: ["bae-6a6482a8-24e1-5c73-a237-ca569e41507d"]) {
        _docID
    }
}
```

```
query {
    Users {
        Name
        _docID
        _version {
            docID
        }
    }
}
```

```
Request: ` {
    commits(groupBy: [docID], order: {docID: DESC}) {
        docID
    }
}`,
Results: []map[string]any{
    {
        "docID": "bae-f54b9689-e06e-5e3a-89b3-f3aee8e64ca7",
    },
    {
        "docID": "bae-72f3dc53-1846-55d5-915c-28c4e83cc891",
    },
},

```

EDIT: Above was resolved with #2162, to do out of this PR.

## Limitations (out of scope of this PR):
- #1467
- #1751
- #1550
- #2162

### CODE REVIEW

File: examples/schema/booksmany.graphql
Comment (shahzadlone) [inline examples/schema/booksmany.graphql:1]: - [ ] Note: This file will be reverted.

File: api/http/errors.go [hunk 1]
```diff
@@ -36,8 +36,6 @@ var (
 	ErrPeerIdUnavailable    = errors.New("no PeerID available. P2P might be disabled")
 	ErrStreamingUnsupported = errors.New("streaming unsupported")
 	ErrNoEmail              = errors.New("email address must be specified for tls with autocert")
-	ErrPayloadFormat        = errors.New("invalid payload format")
```
Comment (shahzadlone) [inline api/http/errors.go:39]: NOTE: Removed becauase were both unused errors.

**Review by fredcarle — COMMENTED** [review]
Definitely going in the right direction. I added a few comments and then looked at how many other reference to dockey there was. There were hundreds so I create a branch (https://github.com/fredcarle/defradb/tree/fredcarle/chore/dockey-rename) with what I found. There might be more changes needed to cover everything.

File: planner/commit.go [hunk 2]
```diff
@@ -351,7 +351,7 @@ func (n *dagScanNode) dagBlockToNodeDoc(block blocks.Block) (core.Doc, []*ipld.L
 	}
 
 	n.commitSelect.DocumentMapping.SetFirstOfName(&commit,
-		request.DockeyFieldName, string(dockey))
+		request.DocID, string(dockey))
```
[the remark this answers is not in the document]
Comment (shahzadlone) [reply planner/commit.go:354]: I do plan to change the local vars.

**Comment by shahzadlone** [comment]
> Definitely going in the right direction. I added a few comments and then looked at how many other reference to dockey there was. There were hundreds so I create a branch ([fredcarle/defradb@`fredcarle`/chore/dockey-rename](https://github.com/fredcarle/defradb/tree/fredcarle/chore/dockey-rename?rgh-link-date=2023-11-24T21%3A21%3A27Z)) with what I found. There might be more changes needed to cover everything.

Yes most of those are local vars and description strings and comments and test function names. Can take care of them too within this PR if there is preference

File: client/request/consts.go [hunk 3]
```diff
@@ -36,14 +32,21 @@ const (
 	OrderClause   = "order"
 	DepthClause   = "depth"
 
+	DocID  = "docID"
+	DocIDs = "docIDs"
+
 	AverageFieldName = "_avg"
 	CountFieldName   = "_count"
-	KeyFieldName     = "_key"
+	KeyFieldName     = "_docID"
```
[the remark this answers is not in the document]
Comment (shahzadlone) [reply client/request/consts.go:40]: Can discuss in other comment

File: client/request/consts.go [hunk 4]
```diff
@@ -98,7 +100,7 @@ var (
 	VersionFields = []string{
 		HeightFieldName,
 		CidFieldName,
-		DockeyFieldName,
+		DocID,
```
[the remark this answers is not in the document]
Comment (fredcarle) [reply client/request/consts.go:103]: I also have a very strong preference for all internal fields to have `_` as it makes it very clear what they are.

File: client/request/consts.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (shahzadlone) [reply client/request/consts.go]: done

**Comment by AndrewSisley** [comment]
thought: I am finding this really hard to effectively review.  There are a lot of independent changes here, and reviewing them all at once with them blended into each other is quite tricky.  This is made harder by the nature of the changes (renaming). 

The commits do not help reviewers as much as they could, as they are not what I would consider complete - where a commit begins a change (e.g. `Rename _key to _dockey`), but leaves the code in a broken state to be patched by later commits (e.g. `Apply naming within unit tests`).  This means as a reviewer I cannot effectively review the rename of _key to _dockey unless I review a guessed range of commits (and hope nothing else is interleaved within that range).

In the future I would suggest either breaking up the PR (I understand why you may want it in one here).  Or broadening the scope of the commits so that reviewers can review a complete, build-passes change in a single commit.

I'm not yet asking for you to change this in this PR, but thought I'd leave the feedback.

**Comment by shahzadlone** [comment]
> thought: I am finding this really hard to effectively review. There are a lot of independent changes here, and reviewing them all at once with them blended into each other is quite tricky. This is made harder by the nature of the changes (renaming).
> 
> The commits do not help reviewers as much as they could, as they are not what I would consider complete - where a commit begins a change (e.g. `Rename _key to _dockey`), but leaves the code in a broken state to be patched by later commits (e.g. `Apply naming within unit tests`). This means as a reviewer I cannot effectively review the rename of _key to _dockey unless I review a guessed range of commits (and hope nothing else is interleaved within that range).
> 
> In the future I would suggest either breaking up the PR (I understand why you may want it in one here). Or broadening the scope of the commits so that reviewers can review a single, complete, build-passes change in a single commit.
> 
> I'm not yet asking for you to change this in this PR, but thought I'd leave the feedback.

I appreciate the feedback and understand the concern, however due to the nature of this PR (renaming). I had to pick between either (1) smaller / consumable / somewhat structured but not clean commits (as the build might fail), OR (2) according to your suggestion range of all commits that form into a clean state as one big commit (i.e. passing build / complete change).

I went with (1) as according to the nature of the PR the review is mainly about weather the renaming makes sense or not, with the assumption that no logic has changed hence it would build. (2) would also in some cases require to change cids multiple times, for individual complete change to pass the build, or like mentioned a really large commit body if I want to change cids once (which to be fair is possible and very easy for me to create even now, I can merge quickly all change commits into one).

Nonetheless will keep this in mind in future for non-rename PRs, I do normally keep commits clean as much as possible.

File: tests/integration/schema/get_schema_test.go
[the remark this answers is not in the document]
Comment (AndrewSisley) [reply tests/integration/schema/get_schema_test.go:1]: Not off the top of my head, I can't remember what order the results we returned in - there is a chance it is sorted by version id atm

**Comment by AndrewSisley** [comment]
> would also in some cases require to change cids multiple times, for individual complete change to pass the build

This is fine, and would quite probably be a net time time saver for the team in a 3000 line PR.

File: cli/cli.go [hunk 5]
```diff
@@ -91,7 +91,7 @@ func NewDefraCommand(cfg *config.Config) *cobra.Command {
 	collection := MakeCollectionCommand(cfg)
 	collection.AddCommand(
 		MakeCollectionGetCommand(),
-		MakeCollectionKeysCommand(),
+		MakeCollectionListDocIDsCommand(),
```
Comment (AndrewSisley) [inline cli/cli.go:94]: question: Why did you add `List` into the func name, and file name?

File: cli/version_test.go [hunk 6]
```diff
@@ -59,11 +59,11 @@ func TestVersionJSON(t *testing.T) {
 	{
 		"release": "",
 		"commit": "",
-		"commitdate": "",
```
Comment (AndrewSisley) [inline cli/version_test.go:62]: question: These look very deliberately lower-case, are you really sure you want to change this?

@nasdf @fredcarle

File: db/base/collection_keys.go [hunk 7]
```diff
@@ -17,18 +17,18 @@ import (
 	"github.com/sourcenetwork/defradb/core"
 )
 
-// MakeIndexPrefix generates a key prefix for the given collection/index descriptions
-func MakeCollectionKey(col client.CollectionDescription) core.DataStoreKey {
+// MakeDSKeyWithCollectionID generates a key prefix for the given collection/index descriptions
```
Comment (AndrewSisley) [inline db/base/collection_keys.go:20]: todo: This documentation is still very broken, I'm sorry as it was broken before, but could you please fix it whilst you are here? It isnt worth a ticket IMO.

Same goes for the function below too (`MakeDocKey`)

Comment (AndrewSisley) [inline db/base/collection_keys.go:20]: todo: Whilst the old name of this function seems odd, the new one doesn't seem very nice either.  I really dislike `DSKey` (instead of `DataStoreKey`).  And `DataStoreKey` itself is out of date and should probably be changed at somepoint (out of scope), meaning adding it to the function name just creates additional places to change when that type rename does happen.

If you really want to include the type name in the constructor name, can you please not abbreviate it, and instead use `DataStoreKey`?

Same change should be applied to the function below too.

**Review by AndrewSisley — CHANGES_REQUESTED** [review]
Some small comments submitted now as I'm context switching mid-review.

File: cli/version_test.go (hunk 6 above)
Comment (shahzadlone) [reply cli/version_test.go:62]: When I first implemented the version stuff (JSON and non-JSON) , I don't remember this being a deliberate hard requirement. Is odd to read all lower IMO.

File: client/collection.go [hunk 8]
```diff
@@ -162,28 +177,28 @@ type Collection interface {
...
 }
 
-// DocKeysResult wraps the result of an attempt at a DocKey retrieval operation.
-type DocKeysResult struct {
-	// If a DocKey was successfully retrieved, this will be that key.
-	Key DocKey
-	// If an error was generated whilst attempting to retrieve the DocKey, this will be the error.
+// DocIDResult wraps the result of an attempt at a DocID retrieval operation.
+type DocIDResult struct {
+	// If a DocID was successfully retrieved, this will be that ID.
```
[the remark this answers is not in the document]
Comment (shahzadlone) [reply client/collection.go:182]: Sure will do.

- [ ] Document DocID not ID

File: client/doc_id.go [hunk 9]
```diff
@@ -20,84 +20,84 @@ import (
...
+	return DocID{
 		version: uint16(version),
 		uuid:    uuid,
 	}, nil
 }
 
-// UUID returns the doc key in UUID form.
-func (key DocKey) UUID() uuid.UUID {
-	return key.uuid
+// UUID returns the DocID in UUID form.
```
[the remark this answers is not in the document]
Comment (shahzadlone) [reply client/doc_id.go:85]: I like your suggestion, will do.

- [ ] document: underlying document identifier

File: client/document.go [hunk 10]
```diff
@@ -479,42 +478,42 @@ func (doc *Document) toMapWithKey() (map[string]any, error) {
...
 	}
-	docMap["_key"] = doc.Key().String()
+	docMap[request.DocIDFieldName] = doc.ID().String()
 
 	return docMap, nil
 }
 
-// GenerateDocKey generates docKey/docID corresponding to the document.
-func (doc *Document) GenerateDocKey() (DocKey, error) {
+// GenerateDocID generates DocID corresponding to the document.
```
[the remark this answers is not in the document]
Comment (shahzadlone) [reply client/document.go:486]: No worries, will add.

- [ ] document: missing `a` word

File: client/errors.go [hunk 11]
```diff
@@ -32,23 +32,16 @@ const (
 // This list is incomplete and undefined errors may also be returned.
 // Errors returned from this package may be tested against these errors with errors.Is.
 var (
```
[the remark this answers is not in the document]
Comment (shahzadlone) [reply client/errors.go:34]: - Yes, all deleted errors were unused. It's because the linter doesn't catch unused public variables when they are become unused by developers, the same way it catches unused private variables.

File: client/request/consts.go [hunk 12]
```diff
@@ -36,14 +32,21 @@ const (
...
-	KeyFieldName     = "_key"
+	DocIDFieldName   = "_docID"
 	GroupFieldName   = "_group"
 	DeletedFieldName = "_deleted"
 	SumFieldName     = "_sum"
 	VersionFieldName = "_version"
 
+	// New generated document id from a backed up document,
+	// which might have a different _docID originally.
+	NewDocIDFieldName = "_docIDNew"
```
[the remark this answers is not in the document]
Comment (shahzadlone) [reply client/request/consts.go:48]: Previously when we had a `_key`, we had a corresponding `_newKey` within backup and recover functionality to mark existing and new keys.

Now with the introduction of a camel case name: `_docID`, I didn't like `_newdocID` for obvious reasons nor `_newDocID` for the simple reason that if we want to do a regex search in the future for all usages of `docID` it won't match.

So I went with `_docIDNew`.

File: db/base/collection_keys.go [hunk 13]
```diff
@@ -17,18 +17,18 @@ import (
...
+func MakeDSKeyWithCollectionID(col client.CollectionDescription) core.DataStoreKey {
 	return core.DataStoreKey{
 		CollectionID: col.IDString(),
 	}
 }
 
-// MakeIndexKey generates a key for the target dockey, using the collection/index description
-func MakeDocKey(col client.CollectionDescription, docKey string) core.DataStoreKey {
+// MakeDSKeyWithCollectionAndDocID generates a DS key for the target docID, using the collection/index description.
```
[the remark this answers is not in the document]
Comment (shahzadlone) [reply db/base/collection_keys.go:27]: I don't mind, and also prefer the extra characters. Will do.

 rename: `MakeDSKeyWithCollectionAndDocID` to `MakeDataStoreKeyWithCollectionAndDocID`

File: db/collection.go [hunk 14]
```diff
@@ -775,48 +774,48 @@ func (c *collection) CreateMany(ctx context.Context, docs []*client.Document) er
...
 	if err != nil {
-		return client.DocKey{}, core.PrimaryDataStoreKey{}, err
+		return client.DocID{}, core.PrimaryDataStoreKey{}, err
 	}
 
-	primaryKey := c.getPrimaryKeyFromDocKey(docKey)
-	if primaryKey.DocKey != doc.Key().String() {
-		return client.DocKey{}, core.PrimaryDataStoreKey{},
-			NewErrDocVerification(doc.Key().String(), primaryKey.DocKey)
+	pdsKey := c.getPrimaryKeyFromDocID(docID)
```
[the remark this answers is not in the document]
Comment (shahzadlone) [reply db/collection.go:785]: I think I was looking to differentiate between Primary Data Store Key vs Data Store Key. Now that I revisit I think primary alone is fine will rename.

- [ ] rename: `pdsKey` to `primaryKey`

File: db/base/collection_keys.go (hunk 7 above)
Comment (shahzadlone) [reply db/base/collection_keys.go:20]: Haha no need to apologize, I agree it's broken and should be improved, will fix in this PR. What specifically do you want me to change that is broken? I am guessing the name `MakeDSKeyWithCollectionID` to `MakeDataStoreKeyWithCollectionID` consistent with your other suggestion?

- [ ] rename: `MakeDSKeyWithCollectionID` to `MakeDataStoreKeyWithCollectionID`

Comment (shahzadlone) [reply db/base/collection_keys.go:20]: Sure so as pointed out in above discussion can do `MakeDataStoreKeyWithCollectionID`:

- [ ] rename: MakeDSKeyWithCollectionID to MakeDataStoreKeyWithCollectionID

Comment (AndrewSisley) [reply db/base/collection_keys.go:20]: `generates` seems like a bad word to use here, as is `key prefix`. `index descriptions` seems quite misleading too.

File: docs/cli/defradb_client_document.md [hunk 15]
```diff
@@ -1,38 +0,0 @@
-## defradb client document
-
-Create, read, update, and delete documents.
```
Comment (AndrewSisley) [inline docs/cli/defradb_client_document.md:3]: question: Why have these files been deleted?

File: cli/cli.go (hunk 5 above)
Comment (shahzadlone) [reply cli/cli.go:94]: Because I was a bit confused what `...CollectionKeysCommand()` actually does. Specially when the others have descriptive actions within their names like `...CollectionUpdateCommand()` and `...CollectionGetCommand()`.

https://github.com/sourcenetwork/defradb/commit/c1eef2081763e4439359a3e88b6191feeba747af#diff-8a4cbfe5ceaae77d6bccca6c6036c192c811898e4392cd40ae7bb8e2e2ce6ae9

From above, it seemed like it actually lists the collection's docIDs hence the name change

File: planner/commit.go [hunk 16]
```diff
@@ -68,15 +68,15 @@ func (n *dagScanNode) Kind() string {
 
 func (n *dagScanNode) Init() error {
 	if len(n.spans.Value) == 0 {
-		if n.commitSelect.DocKey.HasValue() {
-			key := core.DataStoreKey{}.WithDocKey(n.commitSelect.DocKey.Value())
+		if n.commitSelect.DocID.HasValue() {
+			dsKey := core.DataStoreKey{}.WithDocID(n.commitSelect.DocID.Value())
```
Comment (AndrewSisley) [inline planner/commit.go:72]: todo: Related to comments on `DataStoreKey` having an out-of-date name, and worsened by the fact that a `ipfs/datastore.Key` is an actual thing that we use in our codebase, I have quite a strong preference for us to not call anything by ipfs/datastore.Key `dsKey`.

Please rename this variable :sweat_smile: `docID` would probably be fine here.

File: planner/type_join.go [hunk 17]
```diff
@@ -444,9 +444,9 @@ func fetchPrimaryDoc(node, subNode planNode, parentProp string) (bool, error) {
 	if scan == nil {
 		return false, nil
 	}
-	rootDocKey := base.MakeDocKey(scan.col.Description(), docKeyStr)
+	dsKey := base.MakeDSKeyWithCollectionAndDocID(scan.col.Description(), docIDStr)
```
Comment (AndrewSisley) [inline planner/type_join.go:447]: todo: Same as my comment on planner/commit.go - please do not call this `dsKey`

File: db/base/collection_keys.go (hunk 7 above)
Comment (shahzadlone) [reply db/base/collection_keys.go:20]: Sorry I think I edited my comment very close to your last comment, and again now. Let me know if the suggested comment is better. Or if you have another suggestion.

**Review by AndrewSisley — CHANGES_REQUESTED** [review]
Looks good to me, just some small comments added.  Have now reviewed the full PR.

praise: Thank you very much for spreading further the use of the request consts, and cleaning up their names.

Strongly suggest getting in touch with Fred when you are looking to merge this, he has an open PR atm that overlaps a fair amount with this one.

File: db/base/collection_keys.go (hunk 7 above)
Comment (AndrewSisley) [reply db/base/collection_keys.go:20]: Ah yes, happy with your suggestion :)  Thanks Shahzad

File: client/errors.go (hunk 11 above)
[the remark this answers is not in the document]
Comment (AndrewSisley) [reply client/errors.go:34]: Okay, thanks then for deleting them :)

File: db/base/collection_keys.go (hunk 7 above)
Comment (shahzadlone) [reply db/base/collection_keys.go:20]: - [ ] Do suggestion

File: client/request/consts.go (hunk 12 above)
[the remark this answers is not in the document]
Comment (shahzadlone) [reply client/request/consts.go:48]: keeping as is, unless someone has a stronger opinion

File: docs/cli/defradb_client_document.md (hunk 15 above)
Comment (shahzadlone) [reply docs/cli/defradb_client_document.md:3]: These seem outdated docs, do lemme know if anything isn't (my guess is that overtime the `make docs` command generates any new documents, but does not remove old ones).

Comment (AndrewSisley) [reply docs/cli/defradb_client_document.md:3]: These are auto-generated?

File: planner/commit.go (hunk 16 above)
Comment (shahzadlone) [reply planner/commit.go:72]: Cheers. Will rename.

- [ ] rename: `dsKey` -> `docID`

File: planner/type_join.go (hunk 17 above)
Comment (shahzadlone) [reply planner/type_join.go:447]: Will rename.

- [ ] rename: `dsKey` -> `docID`

**Comment by shahzadlone** [comment]
> Looks good to me, just some small comments added. Have now reviewed the full PR.
> 
> praise: Thank you very much for spreading further the use of the request consts, and cleaning up their names.
> 
> Strongly suggest getting in touch with Fred when you are looking to merge this, he has an open PR atm that overlaps a fair amount with this one.

Cheers thanks for reviewing this PR :), for sure will coordinate with @fredcarle

File: docs/cli/defradb_client_document.md (hunk 15 above)
Comment (shahzadlone) [reply docs/cli/defradb_client_document.md:3]: Yea so for example this was regenerated `docs/cli/defradb_client_collection_docIDs.md` with new file name but was originally `docs/cli/defradb_client_collection_keys.md`

And this file that I deleted: `defradb_client_document.md` I am pretty sure has been out dated since `defradb_client_collection.md`: https://github.com/shahzadlone/defradb/blob/gql-renaming-phase1/docs/cli/defradb_client_collection.md

Comment (AndrewSisley) [reply docs/cli/defradb_client_document.md:3]: Ah okay nice - I did not know this :)

I've opened https://github.com/sourcenetwork/defradb/issues/2166 and https://github.com/sourcenetwork/defradb/issues/2165

Comment (shahzadlone) [reply docs/cli/defradb_client_document.md:3]: I didn't either LOL,

Cheers for opening these

File: cli/version_test.go (hunk 6 above)
Comment (fredcarle) [reply cli/version_test.go:62]: I'm ok with the camelCasing.

File: client/request/consts.go (hunk 12 above)
[the remark this answers is not in the document]
Comment (fredcarle) [reply client/request/consts.go:48]: It's just used in backups so it's low impact. I'm ok with `_docIDNew`.

File: db/fetcher/versioned.go [hunk 18]
```diff
@@ -429,7 +429,7 @@ func (vf *VersionedFetcher) Close() error {
 }
 
 // NewVersionedSpan creates a new VersionedSpan from a DataStoreKey and a version CID.
-func NewVersionedSpan(dockey core.DataStoreKey, version cid.Cid) core.Spans {
+func NewVersionedSpan(dsKey core.DataStoreKey, version cid.Cid) core.Spans {
```
Comment (fredcarle) [inline db/fetcher/versioned.go:432]: praise: This was a confusing one before this change. Thanks for fixing!

File: db/collection_delete.go [hunk 19]
```diff
@@ -202,47 +202,47 @@ func (c *collection) deleteWithFilter(
...
 		doc := selectionPlan.Value()
-		// Extract the dockey in the string format from the document value.
-		docKey := doc.GetKey()
 
-		// Convert from string to client.DocKey.
-		key := core.PrimaryDataStoreKey{
+		// Extract the docID in the string format from the document value.
+		docID := doc.GetID()
+
+		pdsKey := core.PrimaryDataStoreKey{
```
Comment (fredcarle) [inline db/collection_delete.go:209]: suggestion: ~~pdsKey~~ primaryDSKey or just primaryKey

File: db/collection_delete.go [hunk 20]
```diff
@@ -202,47 +202,47 @@ func (c *collection) deleteWithFilter(
...
+	results.Count = int64(len(results.DocIDs))
 
 	return results, nil
 }
 
 func (c *collection) applyDelete(
 	ctx context.Context,
 	txn datastore.Txn,
-	key core.PrimaryDataStoreKey,
+	pdsKey core.PrimaryDataStoreKey,
```
Comment (fredcarle) [inline db/collection_delete.go:232]: suggestion: same as above.

File: db/collection_get.go [hunk 21]
```diff
@@ -20,24 +20,24 @@ import (
...
-func (c *collection) Get(ctx context.Context, key client.DocKey, showDeleted bool) (*client.Document, error) {
+func (c *collection) Get(ctx context.Context, docID client.DocID, showDeleted bool) (*client.Document, error) {
 	// create txn
 	txn, err := c.getTxn(ctx, true)
 	if err != nil {
 		return nil, err
 	}
 	defer c.discardImplicitTxn(ctx, txn)
-	dsKey := c.getPrimaryKeyFromDocKey(key)
+	pdsKey := c.getPrimaryKeyFromDocID(docID)
```
Comment (fredcarle) [inline db/collection_get.go:30]: suggestion: same as above.

File: db/collection_get.go [hunk 22]
```diff
@@ -47,7 +47,7 @@ func (c *collection) Get(ctx context.Context, key client.DocKey, showDeleted boo
 func (c *collection) get(
 	ctx context.Context,
 	txn datastore.Txn,
-	key core.PrimaryDataStoreKey,
+	pdsKey core.PrimaryDataStoreKey,
```
Comment (fredcarle) [inline db/collection_get.go:50]: suggestion: same as above.

File: planner/commit.go (hunk 16 above)
Comment (fredcarle) [reply planner/commit.go:72]: it's also not a `docID`. I'm really happy to have changed from `dockey` to `docID` as it removes some of the confusion with a datastore key. This however is not one of them. We us `dsKey` everywhere to represent a `DataStoreKey`. Renaming to `docID` would just flip the confusion from what we had before.

File: planner/type_join.go (hunk 17 above)
Comment (fredcarle) [reply planner/type_join.go:447]: I also oppose this todo. Sorry Andy.

**Review by fredcarle — APPROVED** [review]

**Review by fredcarle — APPROVED** [review]
LGTM! Thanks a lot for doing this Shahzad. It a really nice change and I love that it removes the confusion I sometimes had when not paying too too much attention between dockey and datastore key. Now it's unmistakable.

Please resolve the conversations before merging :)

File: planner/commit.go (hunk 16 above)
Comment (AndrewSisley) [reply planner/commit.go:72]: > We us dsKey everywhere to represent a DataStoreKey

Ah - you are right :) I thought we only used it for `datastore.Key`s :) Leave as-is then :)

Comment (shahzadlone) [reply planner/commit.go:72]: Yea while doing the suggestion I realized the reason I changed it was that the type is not a docID, so will keep as dsKey

File: db/collection_delete.go (hunk 19 above)
Comment (shahzadlone) [reply db/collection_delete.go:209]: - [x] done

File: db/collection_delete.go (hunk 20 above)
Comment (shahzadlone) [reply db/collection_delete.go:232]: - [x] done
